qmd/wide-names.qmd

```diff
@@ -0,0 +1,132 @@
+
+# Wide-names
+
+The exact formula used to generate a wide-name in the ODM depends on the kind of data point you are recording. The three kinds of data and metadata for these column names are either attributes, measures, or protocol steps. An attribute column is metadata. These columns provide context about a measure, protocol, or sample. Measure columns report a specific measure, either from a site, sample, or population. Lastly, protocol columns report a specific protocol step or detail for a measure.
+
+## Attributes
+
+The wide-name formula for attributes is `tableShortName_attribute`.
+
+For example, to record the sample material in long-format you would navigate to the `samples` table and populate the `saMaterial` column. In wide-format, all data are recorded in a single table. The table and attribute are combined for the wide-name, creating the name `sm_saMaterial` for this attribute. Here `sm` is the short name for the samples table, and `saMaterial` is the column name for the sample matrix or material.
+
+## Measures
+
+The wide-name formula for a measure or measure-value is `compartment_specimen_fraction_measure_unit_aggregation_index_attribute`, where the `attribute` at the end is always either `value`, `purpose`, or `qualityFlag`.
+
+The ODM long-format specifies a `measure` column in the `measures` table, as well as a column for `units` and one for `aggregations`. The wide-format collapses all these columns into a single `value` column.
+
+For example, the wide-name for mean chemical oxygen demand from a wastewater sampling site, recorded in milligrams per litre is `wat_si_NR_cod_mgL_m_NR_value`. Here the measure is taken from wastewater/water (`wat`), it's a measure of the site (`si`), the fraction analyzed is not reported (`NR`), it is a measure of chemical oxygen demand (`cod`), the units are mg/L (`mgL`), it's a mean measure (`m`), the index is not reported (`NR`), and lastly this is the value for this measure (`value`).
+
+## Protocol steps
+
+Like measures, protocol step wide-names record the `value` field. With protocols, the wide-name formula depends on whether the step uses a `measure` or a `method`. The formula if using a `method` is: `tableShortName_partTypeShortName_method_attribute`.
+
+The formula if using a `measure` is: `tableShortName_partTypeShortName_measure_unit_aggregation_index_attribute`, where in both instances, the table short name always specifies the `protocolSteps` table (`ps`), and the `partType` short name specifies whether this is a measure (`mes`) or a method (`met`). The attribute is always `value`.
+
+For example, the wide-name for the sequencing strategy protocol step (ie. a protocol step using a method) for a wastewater program would be `ps_met_seqStrat_value`. The protocol steps table short name indicates the long-format table (`ps`), the protocol step uses a method part type (`met`), it reports the sequencing strategy (`seqStrat`), and the column records the value (`value`).
+
```

Comment:
```suggestion
There are various pros and cons to each format, but generally it is ideal for data to be stored and analyzed in a long-format table, while it can be easier to record the data in a wide-format table. Differences in how naming conventions work across formats can lead to mistakes in how the data is mapped, stored and used.
```